container/docker/docker.go

```diff
@@ -0,0 +1,144 @@
+package docker
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/wtsi-ssg/wr/container"
+	"github.com/wtsi-ssg/wr/math/conversion"
+)
+
+type Interactor struct {
+	dockerClient *client.Client
+}
+
+func NewInteractor(cl *client.Client) *Interactor {
+	return &Interactor{
+		dockerClient: cl,
+	}
+}
+
+// ContainerList is a docker specific implementation of the container package function
+// which returns the list of containers.
+func (i *Interactor) ContainerList(ctx context.Context) ([]*container.Container, error) {
+	containerList, err := i.dockerClient.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	customCntrList := make([]*container.Container, len(containerList))
+
+	for idx, cntr := range containerList {
+		customCntrList[idx] = &container.Container{ID: cntr.ID, Names: cntr.Names}
+	}
+
+	return customCntrList, nil
+}
+
+// ContainerStats is a docker specific implementation of the container package function
```

Comment:
Again, could just be "ContainerStats is to implement the Interactor interface.". And have the Interactor interface methods fully documented. So every implementer doesn't have to repeat those docs.

Reply:
Done